### ISSUE TITLE
Question about validation

### ISSUE THREAD

**Issue by hyeok-jong** [rasgally/projects#1]
Thank you for your great work on 'Intelligent Stress Monitoring Assistant for First Responders'. It's impressive that for the first time in WESAD, you introduced the fusion and TCN block. As far as I know, your paper still holds the SOTA performance.
Also your paper is still SOTA performance, as I know.

The code is clean and easy to read. Thank you. 
However, I have a question. For me, it's difficult to read 'tensorflow'-style code, and the version doesn't work well in my environment

I found that in the `model.py` [line](https://github.com/kennethkflai/projects/blob/7222911e0a87c5c4a72d59154c56e8874ec234a5/Stress/util/model.py#L246)
for all the experiments `no_validation = False`.

Thus, the `callbacks` save the model monitoring both 'val_acc' and 'val_out_acc'.
And valid data is actually the test dataset used for LOOSCV.  
(test subject is only one who not included in train dataset)

So, the final model is selected using the test dataset.

Is this correct? I couldn't find accurate information in the paper, so I'm asking.

Again, thanks for your work!

**Comment by kennethkflai** [comment]
1) Due to running LOOSCV, certain subjects are missing some classes, therefore no_validation flag is to check if the number of classes in the training set matches the validation set.  This may not apply to WESAD dataset but some other datasets based on actions/gestures
2) The 2 callbacks based on val_acc and val_out_acc are the default names provided for Windows and Linux systems, respectively.  Usually, only one of them will be used at any one given time.
3) You are correct that the final model is selected using the test dataset.  Depending on the original authors of the dataset, different cross-validations are used.  For WESAD, the comparison is leave-one-subject-out which results in 15 final models (there are 15 subjects) where each final model is trained with 14 subjects and tested on the remaining subject.

**Comment by hyeok-jong** [comment]
Thank you for the kind, fast, and accurate reply. 
I didn't know what the `no_validation ` doing, but now I understand. 
I also didn't know about the 'val_out_acc' at 2, but thank you for letting me know. 
Thank you for checking number 3 once more.
